app/src/main/java/com/example/foody/recipe_details/presentation/RecipeDetailsFragment.kt

```diff
@@ -6,19 +6,32 @@ import android.view.View
 import android.view.ViewGroup
 import androidx.compose.ui.platform.ComposeView
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
 import androidx.navigation.fragment.navArgs
 import com.example.foody.recipe_details.presentation.ui.RecipeDetailsScreen
+import dagger.hilt.android.AndroidEntryPoint
+import dagger.hilt.android.lifecycle.withCreationCallback
 
+@AndroidEntryPoint
 class RecipeDetailsFragment : Fragment() {
     private val args by navArgs<RecipeDetailsFragmentArgs>()
 
+    private val viewModel by viewModels<RecipeDetailsViewModel>(
```

Comment:
The [version of `viewModels` function](https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-master-dev/fragment/fragment-ktx/src/main/java/androidx/fragment/app/FragmentViewModelLazy.kt?source=post_page---------------------------%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F&autodive=0%2F#:~:text=*/-,%40MainThread,%7D,-/**) which takes `extrasProducer` param looks like this:

```Kotlin
@MainThread
public inline fun <reified VM : ViewModel> Fragment.viewModels(
    noinline ownerProducer: () -> ViewModelStoreOwner = { this },
    noinline extrasProducer: (() -> CreationExtras)? = null,
    noinline factoryProducer: (() -> Factory)? = null
): Lazy<VM> {
    val owner by lazy(LazyThreadSafetyMode.NONE) { ownerProducer() }
    return createViewModelLazy(
        VM::class,
        { owner.viewModelStore },
        {
            extrasProducer?.invoke()
                ?: (owner as? HasDefaultViewModelProviderFactory)?.defaultViewModelCreationExtras
                ?: CreationExtras.Empty
        },
        factoryProducer ?: {
            (owner as? HasDefaultViewModelProviderFactory)?.defaultViewModelProviderFactory
                ?: defaultViewModelProviderFactory
        })
}
```